### DISCUSSION TITLE
Does metadata --> max-request not work without uncover?

### DISCUSSION THREAD

**Discussion by snemmani** [projectdiscovery/nuclei-templates#7723] [Q&A] [answered]
I am writing some custom templates that have  a huge list of requests with each request having at least 20 to 30 paths. I would like to know if nuclei works with the below block in the info section of the template. Or should I enable or set some flag for this to come into effect? Because currently, I see nuclei is running and sending all requests in templates which I have marked with this option as well. 

```
metadata:
    max-request: 40
```

I am currently trying to optimize the number of requests sent and `stop-at-first-match: true` is one such option I am using.
Any inputs in doing this efficiently are appreciated. 

Thanks in Advance!

**Comment by princechaddha** [comment, marked as answer]
hello @snemmani, Thank you for taking the time to create this discussion

The fields within the metadata primarily serve to provide additional information for the template. The max-requests field indicates the maximum number of requests that will be sent when running a specific template. However, it does not impose any limits on the number of requests sent at the engine level. As for stop-at-first-match, it only comes into effect if an endpoint is matched, in which case, the other requests from the templates will not be sent. Otherwise, all the requests will be sent.

Currently, we do not have any functionality to halt requests at the engine level if they are included in the templates. If you believe this would be a valuable addition to the engine, please feel free to raise a nuclei issue [here](http://github.com/projectdiscovery/nuclei).  You can join our [discord server](https://discord.com/invite/projectdiscovery). It's a great place to connect with fellow contributors and stay updated with the latest developments. Thank you once again

**Comment by snemmani** [comment]
Issue created https://github.com/projectdiscovery/nuclei-templates/issues/7763
